### ISSUE TITLE
Fix maven OSGI warning in 2.0.x

### ISSUE THREAD

**Issue by sbernard31** [eclipse/californium#563]
I cleaned the maven build but there is still some warning about OSGI.

```
[WARNING] Bundle org.eclipse.californium:scandium:bundle:2.0.0-SNAPSHOT : Export org.eclipse.californium.scandium.dtls,  has 1,  private references [org.eclipse.californium.scandium.util], 
[WARNING] Bundle org.eclipse.californium:scandium:bundle:2.0.0-SNAPSHOT : Export org.eclipse.californium.scandium.dtls.pskstore,  has 1,  private references [org.eclipse.californium.scandium.util], 
[WARNING] Bundle org.eclipse.californium:californium-core:bundle:2.0.0-SNAPSHOT : Export org.eclipse.californium.core.network,  has 1,  private references [org.eclipse.californium.core.network.deduplication], 
WARNING] Bundle org.eclipse.californium:californium-proxy:bundle:2.0.0-SNAPSHOT : Export org.eclipse.californium.proxy.resources,  has 1,  private references [org.eclipse.californium.compat], 
```

I didn't look so much of that but it seems relevant to me.
So I create this issue to not forget to have a deeper look about that.

**Comment by sbernard31** [comment]
I looked at this a bit deeper.

We currently use `maven-bundle-plugin` to generate OSGI ready jar.
In OSGI you must export package to make it usable by other bundle. (so if you don't export it, it will only be used internally)

This 4 warnings is about classes exported publicly which expose classes (via method argument for example)  which is not exported (private package).

From an OSGI point of view it's a real problem.

Should we still support OSGI ?
If no, 
1) we can just remove all maven-bundle-plugin configuration.
If yes, there is several way to fix it.
2.1) just change current private package to public. (if we would not infest so mush in OSGI we could even consider to make all package public)
2.2) make a proper separation between private/public package, generally we create package like ${project.artifactId}.impl or ${project.artifactId}.internal for all private package and the rest is public (this will probably bring more question discussion later)

I have no strong opinion about that, but I would go for 1) or 2.1) as we are not so sure OSGI is needed/used.

(classes which should be made public : Deduplicator, ServerName, ServerNames and the whole org.eclipse.californium.compat package)

**Comment by sophokles73** [comment]
I am aware of at least one party interested in using Cf in OSGi so FMPOV we should continue to support OSGi. I do agree with your analysis regarding the cause of the problem, @sbernard31. I also think that we should go for 2.1. In fact, the package design in Californium core has been dubious from the beginning AFAIC, so that there is no clear *direction of dependency* and thus no easy to identify separation between public and private packages. So, I am with you in that we should simply export all packages.

**Comment by sbernard31** [comment]
Fixed by #608
